### ISSUE TITLE
SRP - Captured Output regex 

### ISSUE THREAD

**Issue by jamesladd** [TheSecretSquad/toyrobotsimulator#4]
The ToyRobotSimulatorTest has methods to help with pattern matching and constructing a Pattern. These don't belong in the Test and should instead be in a separate class similar to how CapturingPrintStream is used. This will make the Test size smaller.

Maybe even have a Matcher that allows you to match with or without a newline and thus remove
additional methods that are about matching from the Test.

**Comment by TheSecretSquad** [comment]
Ok :smiley_cat: :+1:

**Comment by TheSecretSquad** [comment]
@jamesladd Should I write tests for the "matching" code? Seems weird to write tests for something that's used to test, but what guarantees that it's working correctly and not corrupting my tests of the simulation?

Come to think of it. Instead of the CapturingPrintStream having a getter on it for the contents, couldn't I put the matching methods on CapturingPrintStream and have it just verify its own contents?

**Comment by jamesladd** [comment]
Yes, this is a support class like capturing stream - write a test

**Comment by TheSecretSquad** [comment]
@jamesladd Thanks!
